core-processor/src/ext.rs

```diff
@@ -474,7 +726,40 @@ impl BackendExternalities for Ext {
 }
 
 impl Ext {
-    fn check_message_value(&mut self, message_value: u128) -> Result<(), FallibleExtError> {
+    fn with_changes<F, R>(&mut self, callback: F) -> Result<R, FallibleExtError>
```

Comment:
seems we can introduce generic impl for `with_changes` by abstracting error type, can't we?